bst.py

```diff
@@ -0,0 +1,272 @@
+                node = node.left
+            else:
+                node = stack.pop()
+                node = node.right
+
+    def post_order(self):
+        """Return a generator with tree values from post-order traversal"""
+        stack = []
+        node = self
+        last = None
+        while stack or node:
+            if node:
+                stack.append(node)
+                node = node.left
+            else:
+                peek = stack[-1]
+                if peek.right is not None and last != peek.right:
+                    node = peek.right
+                else:
+                    yield peek.val
+                    last = stack.pop()
+                    node = None
+
+    def breadth_first(self):
+        """Return a generator with tree values from breadth first traversal"""
+        q = Queue()
+        q.enqueue(self)
+        while q.size() > 0:
+            node = q.dequeue()
+            yield node.val
+            if node.left:
+                q.enqueue(node.left)
+            if node.right:
+                q.enqueue(node.right)
+
+    def get_dot(self):
+        """Return the tree with root as a dot graph for visualization."""
+        return "digraph G{\n%s}" % ("" if self.val is None else (
+            "\t%s;\n%s\n" % (
+                self.val,
+                "\n".join(self._get_dot())
+            )
+        ))
+
+    def _get_dot(self):
+        """recursively prepare a dot graph entry for this node."""
+        if self.left is not None:
+            yield "\t%s -> %s;" % (self.val, self.left.val)
+            for i in self.left._get_dot():
+                yield i
+        elif self.right is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.val, r)
+        if self.right is not None:
+            yield "\t%s -> %s;" % (self.val, self.right.val)
+            for i in self.right._get_dot():
+                yield i
+        elif self.left is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.val, r)
+
+    def save_render(self, savefile="tree.gv"):
+        """Render and save a represntation of the tree.
+
+        args:
+            savefile: the optional filename
+        """
+        from graphviz import Source
+        src = Source(self.get_dot())
+        path = 'graphviz/{}'.format(savefile)
+        src.render(path)
+
+    @classmethod
+    def _sorted_list_to_BST(cls, items=[], start=None, end=None):
+        """Create a balanced binary search tree from sorted list.
+
+        args:
+            items: the sorted list of items to insert into tree
+            start: the start of the list
+            end: the end of the list
+
+        returns: a balanced binary search tree (node)
+        """
+        if start > end:
+            return None
+        mid = start + (end - start) / 2
```

Comment:
Watch out for this in Python 3.  There, true division will return a float even if both arguments are integers.  What you really want here is *floored* division, the `//` operator